File: app/src/main/java/com/wishnewjam/dubstepfm/ChooseBitrateDialogFragment.java
package com.wishnewjam.dubstepfm;

import android.app.Activity;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

public class ChooseBitrateDialogFragment extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle(R.string.choose_bitrate)
                .setItems(R.array.bitrates, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int which) {
                        changeBitrate(which);
                    }
                });
        return builder.create();
    }

    private void changeBitrate(int which) {
        PreferenceManager.getDefaultSharedPreferences(getActivity())
                .edit().putString(MainService.SP_KEY_BITRATE, Links.LINK_24).apply();
        switch (which) {
            case 0:
                PreferenceManager.getDefaultSharedPreferences(getActivity())
                        .edit().putString(MainService.SP_KEY_BITRATE, Links.LINK_24).apply();
                dismiss();
                break;
            case 1:
                PreferenceManager.getDefaultSharedPreferences(getActivity())
                        .edit().putString(MainService.SP_KEY_BITRATE, Links.LINK_64).apply();
                dismiss();
                break;
            case 2:
                PreferenceManager.getDefaultSharedPreferences(getActivity())
                        .edit().putString(MainService.SP_KEY_BITRATE, Links.LINK_128).apply();
                dismiss();
                break;
            case 3:
                PreferenceManager.getDefaultSharedPreferences(getActivity())
                        .edit().putString(MainService.SP_KEY_BITRATE, Links.LINK_256).apply();
                dismiss();
                break;
        }
    }

    @Override
    public void onDismiss(DialogInterface dialog) {
        super.onDismiss(dialog);
        final Activity activity = getActivity();
        if (activity instanceof DialogInterface.OnDismissListener) {
            ((DialogInterface.OnDismissListener) activity).onDismiss(dialog);
        }
    }
}

File: app/src/main/java/com/wishnewjam/dubstepfm/ChooseBitrateDialogFragment.kt
package com.wishnewjam.dubstepfm

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.TextView
import androidx.core.content.res.ResourcesCompat
import androidx.core.net.toUri
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders

class ChooseBitrateDialogFragment : androidx.fragment.app.DialogFragment(),
        View.OnClickListener {

    private var items: ArrayList<TextView> = ArrayList()
    private lateinit var mediaViewModel: MediaViewModel

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val v = inflater.inflate(R.layout.dialogfragment_bitrate, container,
                false)
        items = arrayListOf(v.findViewById(R.id.tv_bitrate_24), v.findViewById(R.id.tv_bitrate_64), v.findViewById(R.id.tv_bitrate_128),
            v.findViewById(R.id.tv_bitrate_256))
        for (item in items) {
            item.setOnClickListener(this)
        }
        mediaViewModel = ViewModelProviders.of(this)
                .get(MediaViewModel::class.java)
        // TODO:
        // mediaViewModel.currentUrl.observe(this,
        //         Observer<String> { t -> t?.let { colorize(it) } })

        v.findViewById<CheckBox>(R.id.chb_consent).isChecked = mediaViewModel.userConsent.value ?: true
        v.findViewById<CheckBox>(R.id.chb_consent).setOnCheckedChangeListener { _, isChecked ->
            mediaViewModel.changeConsent(isChecked)
        }

        v.findViewById<TextView>(R.id.tv_privacy_policy).setOnClickListener {
            startActivity(
                    Intent(Intent.ACTION_VIEW, Links.PRIVACY_POLICY.toUri()))
        }
        return v
    }

    override fun onClick(p0: View?) {
        when (p0?.id) {
            R.id.tv_bitrate_24 -> {
                changeBitrate(0)
            }
            R.id.tv_bitrate_64 -> {
                changeBitrate(1)
            }
            R.id.tv_bitrate_128 -> {
                changeBitrate(2)
            }
            R.id.tv_bitrate_256 -> {
                changeBitrate(3)
            }
        }
    }

    private fun changeBitrate(which: Int) {
        val br: String = Links.AS_ARRAY[which]
        mediaViewModel.changeBitrate(br)
        dismiss()
    }

    private fun colorize(which: String) {
        colorize(Links.AS_ARRAY.indexOf(which))
    }

    private fun colorize(which: Int) {
        for ((index, value) in items.withIndex()) {
            if (index == which) {
                value.setTextColor(ResourcesCompat.getColor(resources,
                        R.color.text_color_secondary, null))
            }
            else {
                value.setTextColor(Color.WHITE)
            }
        }
    }
}
